homeassistant/components/totalconnect/alarm_control_panel.py

```diff
@@ -193,6 +193,10 @@ def _arm_home(self):
         """Arm home synchronous."""
         try:
             ArmingHelper(self._partition).arm_stay()
+        except UsercodeInvalid as error:
+            raise ConfigEntryAuthFailed(
+                "TotalConnect usercode is invalid. Did not arm home."
+            ) from error
```

Comment:
I'm not sure this will work when called from a service

Reply:
So i need to do something like below ?

```python
        self.hass.async_create_task(
            self.hass.config_entries.flow.async_init(
                DOMAIN,
                context={"source": SOURCE_REAUTH},
            )
        )
```

Reply:
You could call `entry.async_start_reauth()`

https://github.com/home-assistant/core/blob/dev/homeassistant/config_entries.py#L616

Reply:
Thanks moving to that solution.  Would be nice if `CoordinatorEntity` provided such a function so the call could just be
`self.async_start_reauth()`
instead of
`self.coordinator.config_entry.async_start_reauth(self.hass)`

....I suppose I'm tasking myself with a new PR